assets/wire/utils.ts

```diff
@@ -259,28 +245,27 @@ export function shortHighlightedtext(html: string, maxLength = 40) {
     const div = document.createElement('div');
     div.innerHTML = html;
     const highlightSpans = Array.from(div.querySelectorAll('span.es-highlight'));
-
-    let extractedText = '';
+    const extractedText: string[] = [];
+    let lastWord = '';
 
     for (const highlightSpan of highlightSpans) {
-        const previousNode = highlightSpan.previousSibling;
-        if (previousNode && previousNode.nodeType === Node.TEXT_NODE) {
-            const {text, remainingCount} = getTextFromNode(previousNode, maxLength);
-            extractedText += text;
-            maxLength = remainingCount;
+        const textBefore = getTextFromNode(highlightSpan.previousSibling);
+        const textAfter = getTextFromNode(highlightSpan.nextSibling);
+
+        if (textBefore && textBefore !== lastWord) {
+            extractedText.push(textBefore);
+            lastWord = textBefore;
         }
 
-        extractedText += highlightSpan.outerHTML;
+        extractedText.push(highlightSpan.outerHTML);
 
-        const nextNode = highlightSpan.nextSibling;
-        if (nextNode && nextNode.nodeType === Node.TEXT_NODE) {
-            const {text, remainingCount} = getTextFromNode(nextNode, maxLength);
-            extractedText += text;
-            maxLength = remainingCount;
+        // Append text after the highlight span
+        if (textAfter && textAfter !== lastWord) {
```

Comment:
not sure if we should compare it to `lastWord` here because that would probably skip it in eg.:
```html
<p>foo <span class="es...">bar</span> foo <span class="es...">bar</span> foo</p>
```
would become:
```html
<p>foo <span ...>bar</span> <span ...>bar</span></p>
```
but maybe not? might be something for a test case
in general you want to avoid using `textBefore` from node if that very same node was `nextSibling`
so rather than checking for last word it might be better to store last node

Reply:
sounds identical